go/vt/mysqlctl/backup.go

```diff
@@ -334,6 +334,16 @@ func Restore(ctx context.Context, params RestoreParams) (*BackupManifest, error)
 		return nil, err
 	}
 
+	// We disable super_read_only, in case it is in the default MySQL startup
+	// parameters and will be blocking the writes we need to do in
+	// PopulateMetadataTables().  We do it blindly, since
+	// this will fail on MariaDB, which doesn't have super_read_only
+	// This is safe, since we're restarting MySQL after the restore anyway
+	params.Logger.Infof("Restore: disabling super_read_only")
+	if err := params.Mysqld.SetSuperReadOnly(false); err != nil {
```

Comment:
It will be better to check for a specific error to ignore for mariadb (and log warning and move on), and fail the operation if there is some other unexpected error.

Reply:
One of the problems here is that it is not trivial to know if it is MariaDB or not, without writing a whole set of version parsing code;  I was initially going to do that and then just try to set super_read_only for non-mariadb.  I'll do a best-effort error without having to dig that out.